### PR TITLE
JAMES-3544 Attachment refactoring (step 2)

### PR DESCRIPTION
**Pull request by chibenwa**
- Hold messageId within mail table
 - Rework JMAP draft to reduce queries related to attachments
 - Migration to add messageId to attachment table
 - Group reads on the BlobManager - this allows querying only once the message/mailbox context of attachments

### CODE REVIEW

File: server/protocols/jmap-draft-integration-testing/memory-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/memory/MemorySpamAssassinContractTest.java [hunk 1]
```diff
@@ -1,82 +0,0 @@
-/****************************************************************
```
Comment (chibenwa) [inline server/protocols/jmap-draft-integration-testing/memory-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/memory/MemorySpamAssassinContractTest.java:1]: Oups

File: server/protocols/jmap-draft-integration-testing/memory-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/memory/MemoryVacationRelayIntegrationTest.java [hunk 2]
```diff
@@ -1,47 +0,0 @@
-/****************************************************************
```
Comment (chibenwa) [inline server/protocols/jmap-draft-integration-testing/memory-jmap-draft-integration-testing/src/test/java/org/apache/james/jmap/memory/MemoryVacationRelayIntegrationTest.java:1]: Oups

**Review by Arsnael — APPROVED** [review]

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/AttachmentMessageIdMigration.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    @Override
+    public void apply() {
+        attachmentMessageIdDAO.listAll()
+            .flatMap(this::migrate, DEFAULT_CONCURRENCY)
+            .blockLast();
+    }
+
+    private Mono<Void> migrate(Pair<AttachmentId, MessageId> entry) {
+        return attachmentDAO.insertMessageId(entry.getKey(), entry.getValue());
```
Comment (vttranlina) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/AttachmentMessageIdMigration.java:100]: Is it possible that `AttachmentId` is ready in `attachmentDAO` ?

Comment (chibenwa) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/AttachmentMessageIdMigration.java:100]: Yes it is possible that AttachmentId is already in attachmentDAO, in such cases we overwrite it.

And it is also perfectly fine performance-wise: better do extra writes that extra reads according to Cassandra phylosophy...

**Comment by chibenwa** [comment]
Force pushed to solve conflict

**Pull request merged**